### PR TITLE
Add Google Attestation

### PR DESCRIPTION
**Pull request by shahthepro**
Adds Google Attestation. #1826

### CODE REVIEW

**Comment by nick** [comment]
Seems there's 1 failing test:

```
google attestations
    ✓ should generate a correct auth url
    x should generate attestation on valid verification code
    ✓ should error on missing verification code
```

**Comment by shahthepro** [comment]
@nick Thanks. Have fixed the unit test and made those changes. :)

**Comment by franckc** [comment]
Awesome ! Thanks @shahthepro for implementing this :)
Looking really good. Nice work.

Couple comments:
 - Let's store whether the user is Google verified in the identity table. Could you add a column in this table [here](https://github.com/OriginProtocol/origin/blob/master/infra/identity/src/models/identity.js#L14) and make sure it is populated by the listener [here](https://github.com/OriginProtocol/origin/blob/master/infra/discovery/src/listener/handler_identity.js#L132) ?
- Before we land this we need to figure what's the impact on Origin Rewards. Should users get rewarded for the verification of their Google account the same way as for other attestations ? And shall that be part of April campaign (e.g. we would need to update the campaign rules) or should it start in May campaign ? @matthewliu could you comment on this please ?

**Comment by shahthepro** [comment]
@franckc 

> Awesome ! Thanks @shahthepro for implementing this :)
> Looking really good. Nice work.

Welcome :)

> Let's store whether the user is Google verified in the identity table. Could you add a column in this table here and make sure it is populated by the listener here?

I somehow overlooked it. Have made that change.

**Comment by shahthepro** [comment]
@franckc will make that change :)

**Comment by shahthepro** [comment]
@franckc Have made that change

File: dapps/marketplace/src/pages/user/Profile.js [hunk 1]
```diff
@@ -186,7 +186,8 @@ class UserProfile extends Component {
                 )}
                 {this.renderAtt(
                   'google',
-                  fbt('Google', '_ProvisionedChanges.google')
+                  fbt('Google', '_ProvisionedChanges.google'),
+                  process.env.ENABLE_GOOGLE_ATTESTATION != 'true'
```
Comment (franckc) [inline dapps/marketplace/src/pages/user/Profile.js]: nit: in our codebase we use the !== operator rather than !=

**Comment by franckc** [comment]
Thanks @shahthepro ! Minor nit inline otherwise looks great to me.
@nick if the DApp code looks good to you, let's merge this to master :)

Before we deploy to staging/prod, we'll need to add the following keys to EnvKey:
 - DApp: ENABLE_GOOGLE_ATTESTATION
 - Bridge: GOOGLE_CLIENT_ID and GOOGLE_CLIENT_SECRET
As our secrets guardian, perhaps @joshfraser could be help with creating and setting GOOGLE_CLIENT_ID and GOOGLE_CLIENT_SECRET ?

**Pull request merged**